src/Controls/src/Core/Platform/Windows/CollectionView/ItemContentControl.cs

```diff
@@ -163,12 +163,19 @@ internal void Realize()
 				return;
 			}
 
+			// If the template is a DataTemplateSelector, we need to get the actual DataTemplate to use
+			// for our recycle check below
+			if (formsTemplate is DataTemplateSelector dataTemplateSelector)
```

Comment:
Can we use this extension method that should probably be internal?

https://github.com/dotnet/maui/blob/eb9666b5ed558f4bc6a5d7956c84e160e49c3a30/src/Controls/src/Core/DataTemplateExtensions.cs#L11-L18

iOS uses this extension method to realize its templates as well